src/parser.c

```diff
@@ -41,6 +41,15 @@ void initFixed(struct FixedState *const state, size_t const len) {
     memset(&state->buffer, 0, len);
 }
 
+enum transaction_type_id_t convert_type_id_to_type(uint32_t type_id) {
+  switch (type_id) {
+      case 0: return TRANSACTION_TYPE_ID_BASE;
+      case 3: return TRANSACTION_TYPE_ID_IMPORT;
+      case 4: return TRANSACTION_TYPE_ID_EXPORT;
```

Comment:
Pretty sure these can be `case TRANSACTION_TYPE_ID_EXPORT: return TRANSACTION_TYPE_ID_EXPORT`.  Any particular reason to have a literal integer on the left hand side?

Reply:
Sure: While i agree, type_id is explicitly a uint32_t. This function makes the conversion explicit, and I think I'd prefer (although not *too* strongly) to keep this the way it is, as it adds an extra "check" against getting the number wrong